backend/post-service/src/main/java/com/speakapp/postservice/services/PostService.java

```diff
@@ -56,6 +56,33 @@ public PostGetDTO createPost(PostCreateDTO postCreateDTO, UUID userId) {
         );
     }
 
+    public PostGetDTO updatePost(PostCreateDTO postCreateDTO, UUID postId, UUID userId) {
+        Post postUpdated = postRepository.getPostByPostId(postId);
+
+        //TODO: Handling Exception when userId != postId
+
+        postMapper.updatePostFromPostCreateDTO(postCreateDTO, postUpdated);
```

Comment:
Tutaj właśnie będzie problem przez tego nulla, tzn. jeżeli nie znajdziemy obiektu w bazie z podanym postId, to w tym miejscu próbujesz zmapować wartości na obiekt null -> NullPointerException